pandas/plotting/_matplotlib/core.py

```diff
@@ -1345,6 +1351,16 @@ def _make_plot(self):
 
         for i, (label, y) in enumerate(self._iter_data(fillna=0)):
             ax = self._get_ax(i)
+
+            if self.orientation == "vertical":
+                ax.xaxis.update_units(self.ax_index)
+                self.tick_pos = ax.convert_xunits(self.ax_index).astype(np.int)
+                self.ax_pos = self.tick_pos - self.tickoffset
+            elif self.orientation == "horizontal":
+                ax.yaxis.update_units(self.ax_index)
+                self.tick_pos = ax.convert_yunits(self.ax_index).astype(np.int)
+                self.ax_pos = self.tick_pos - self.tickoffset
```

Comment:
i think this `self.ax_pos = self.tick_pos - self.tickoffset` can be taken out from `if.. else..` clause, because the main purpose of this clause is to get `self.tick_pos` for bar and barh separately.

Reply:
Sure thing.